packages/browser-wallet/src/background/identity-issuance.ts

```diff
@@ -12,85 +12,106 @@ import { confirmIdentity } from './confirmation';
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=1024211#c73
+    chrome.alarms.clear(IDP_ALARM_NAME);
 
-async function startIdentityIssuance({
-    baseUrl,
-    ...identityRequestInputs
-}: IdentityRequestInput & { baseUrl: string }) {
     const network = await storedCurrentNetwork.get();
     if (!network) {
         return;
     }
-    const respond = async (response: IdentityIssuanceBackgroundResponse) => {
-        let { status } = response;
-        const pending = await sessionPendingIdentity.get();
-        if (!pending) {
-            status = BackgroundResponseStatus.Aborted;
-        } else {
-            if (response.status === BackgroundResponseStatus.Success) {
-                const newIdentity: PendingIdentity = {
-                    ...pending,
-                    status: CreationStatus.Pending,
-                    location: response.result,
-                };
-                await addIdentity(newIdentity, network.genesisHash);
-                confirmIdentity(newIdentity, network.genesisHash);
-            }
-            await sessionPendingIdentity.remove();
+
+    let { status } = response;
+    const pending = await sessionPendingIdentity.get();
+
+    if (!pending) {
+        status = BackgroundResponseStatus.Aborted;
+    } else {
+        if (response.status === BackgroundResponseStatus.Success) {
+            const newIdentity: PendingIdentity = {
+                ...pending,
+                status: CreationStatus.Pending,
+                location: response.result,
+            };
+            await addIdentity(newIdentity, network.genesisHash);
+            confirmIdentity(newIdentity, network.genesisHash);
+        }
+        await sessionPendingIdentity.remove();
+    }
+
+    await openWindow();
+    bgMessageHandler.sendInternalMessage(InternalMessageType.EndIdentityIssuance, { ...response, status });
+};
+
+export function addIdpListeners() {
+    const closedListener = async (tabId: number) => {
+        const idpTabId = await sessionIdpTab.get();
+
+        if (idpTabId !== undefined && tabId === idpTabId) {
+            chrome.tabs.onRemoved.removeListener(closedListener);
+            respond({
+                status: BackgroundResponseStatus.Aborted,
+            });
         }
-        await openWindow();
-        bgMessageHandler.sendInternalMessage(InternalMessageType.EndIdentityIssuance, { ...response, status });
     };
 
+    chrome.tabs.onRemoved.addListener(closedListener);
+
+    const handleIdpRequest = (redirectUrl: string, tabId: number) => {
+        chrome.tabs.onRemoved.removeListener(closedListener);
+
+        if (tabId !== undefined) {
+            chrome.tabs.remove(tabId);
+            sessionIdpTab.remove();
+        }
+
+        respond({
+            status: BackgroundResponseStatus.Success,
+            result: redirectUrl.substring(redirectUrl.indexOf(codeUriKey) + codeUriKey.length),
+        });
+    };
+
+    chrome.webRequest.onBeforeRequest.addListener(
```

Comment:
No more on redirect listener? Have you checked that we don't need that for Notabene and DTS?

Reply:
Yes, they all work. Redirect triggers onBeforeRequest as well.